presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1086,6 +1086,12 @@ private void prepareAddTable(HdfsContext context, TableAndMore tableAndMore)
             deleteOnly = false;
 
             Table table = tableAndMore.getTable();
+
+            if (table.isTemporary()) {
```

Comment:
Today all the use case for temporary table is insert existing table right?

Reply:
Yes. The semantic is always CREATE + INSERT

Reply:
This is needed to cover a hypotetical corner case when a temporary table is created, but never inserted.